src/aspire/abinitio/commonline_utils.py

```diff
@@ -100,95 +130,91 @@ def estimate_inplane_rotations(cl_class, vis):
 
     # Normalize rays.
     pf /= norm(pf, axis=-1)[..., np.newaxis]
 
     n_pairs = n_img * (n_img - 1) // 2
-    with tqdm(total=n_pairs) as pbar:
-        idx = 0
-        # Note: the ordering of i and j in these loops should not be changed as
-        # they correspond to the ordered tuples (i, j), for i<j.
-        for i in range(n_img):
-            pf_i = pf[i]
-
-            # Generate shifted versions of images.
-            pf_i_shifted = np.array(
-                [pf_i * shift_phase for shift_phase in shift_phases]
-            )
-
-            Ri_tilde = Ri_tildes[i]
-
-            for j in range(i + 1, n_img):
-                pf_j = pf[j]
+    pbar = tqdm(total=n_pairs)
+    # Note: the ordering of i and j in these loops should not be changed as
+    # they correspond to the ordered tuples (i, j), for i<j.
+    for i in range(n_img):
+        pf_i = pf[i]
 
-                Rj_tilde = Ri_tildes[j]
+        # Generate shifted versions of images.
+        pf_i_shifted = np.array([pf_i * shift_phase for shift_phase in shift_phases])
 
-                # Compute all possible rotations between the i'th and j'th images.
-                Us = np.array(
-                    [Ri_tilde.T @ R_theta_ij @ Rj_tilde for R_theta_ij in R_theta_ijs]
-                )
+        Ri_tilde = Ri_tildes[i]
 
-                # Find the angle between common lines induced by the rotations.
-                c1s = np.array([[-U[1, 2], U[0, 2]] for U in Us])
-                c2s = np.array([[U[2, 1], -U[2, 0]] for U in Us])
+        for j in range(i + 1, n_img):
+            pf_j = pf[j]
 
-                # Convert from angles to indices.
-                c1s = cl_angles_to_ind(c1s, n_theta)
-                c2s = cl_angles_to_ind(c2s, n_theta)
+            Rj_tilde = Ri_tildes[j]
 
-                # Perform correlation, corrs is shape n_shifts x len(theta_ijs).
-                corrs = np.array(
-                    [
-                        np.dot(pf_i_shift[c1], np.conj(pf_j[c2]))
-                        for pf_i_shift in pf_i_shifted
-                        for c1, c2 in zip(c1s, c2s)
-                    ]
-                )
+            # Compute all possible rotations between the i'th and j'th images.
+            Us = np.array(
+                [Ri_tilde.T @ R_theta_ij @ Rj_tilde for R_theta_ij in R_theta_ijs]
+            )
 
-                # Reshape to group by shift and symmetric order.
-                corrs = corrs.reshape((n_shifts, order, len(theta_ijs) // order))
+            # Find the angle between common lines induced by the rotations.
+            c1s = np.array([[-U[1, 2], U[0, 2]] for U in Us])
+            c2s = np.array([[U[2, 1], -U[2, 0]] for U in Us])
+
+            # Convert from angles to indices.
+            c1s = _cl_angles_to_ind(c1s, n_theta)
+            c2s = _cl_angles_to_ind(c2s, n_theta)
+
+            # Perform correlation, corrs is shape n_shifts x len(theta_ijs).
+            corrs = np.array(
```

Comment:
This would probably be clearer without the nested list comprehension. And you might not need to reshape it ....